### ISSUE TITLE
Stack depth very limited

### ISSUE THREAD

**Issue by shriram** [vishesh/racketscript#25]
I find that when running a tight non-tail loop, the stack exhausts in around 12,000 frames on my desktop browser, which is just a second or two of execution time. This makes it difficult to run any interesting recursive computations. When will the project add stack management?

**Comment by stchang** [comment]
Yes, the stack size will depend on the browser. By stack management, do you mean continuations?

**Comment by shriram** [comment]
By stack management I mean not being bounded by the limitations of
browsers, some of which (especially on mobile) have VERY shallow stacks.

I'd be using this to program in Racket — if I wanted to program in
JavaScript I'd use JavaScript directly. (-:

**Comment by stchang** [comment]
> I'd be using this to program in Racket — if I wanted to program in JavaScript I'd use JavaScript directly. (-:

I don't buy the all-or-nothing dichotomy. The success of languages like ClojureScript, BuckleScript, and others suggests that there is a middle ground worth exploring.

Racket has many great features and the main goal of this project is to experiment with what we can compromise on, and what benefits we can get in return for making such tradeoffs.

The documentation definitely needs to be more precise about this though.
